src/backend/parser/parse_func.c

```diff
@@ -2059,9 +2059,22 @@ funcname_signature_string(const char *funcname, int nargs,
 	int			numposargs;
 	ListCell   *lc;
 	int			i;
+	const char *ptr;
 
 	initStringInfo(&argbuf);
 
+	/* Check if there is schema in funcname */
+	while (ptr) {
+		if (*ptr == '.') {
+            break;
+        }
+        ptr++;
+	}
```

Comment:
To minimize change in engine code, please move this logic into hook func. 
Also, I searched in BBF code base and found a function `split_object_name` that splits multiple-part name into an array. There should be some examples of using it in BBF repo.
In test cases, please also consider delimiters, upper/lower/mixed cases.